build-support/bin/release-changelog-helper.sh

```diff
@@ -57,10 +61,6 @@ Refactoring, Improvements, and Tooling
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 
-Testing
```

Comment:
I don't get the argument that this is covered by other categories?

Reply:
I've never understood what it's meant to be in the first place?

If we make changes to `pants.testutil`, those go into "Plugin changes". 

If we make changes to our own tests, why would that go into a dedicated section rather than "Refactoring"? This changelog is intended for end-users, not as much for us. They don't really care if we refactor some tests.

Reply:
I think it's good to call out testing improvements, such as adding tests. Many changes don't affect end-users directly, but they still need to be in some category or another.

Reply:
> Many changes don't affect end-users directly, but they still need to be in some category or another.

Which is why we have the "Refactoring, Internal Tooling" category. I'm not sure why changes to tests deserve to be elevated over changes to production code and over changes to our build process/CI? From an end-user perspective, they are all "internal changes" to Pants that make the project better, without directly impacting the user.

To be pedantic, if we kept "Testing", I think we would want to split out "Internal changes" into something like "Test internal changes", "Production internal change", and "Build process internal change". Which I suspect we'd agree we don't want to do; that's overspecified for a changelog.

Reply:
+1 I think a changelog is fundamentally for users. Iff you decide to log every change, then you're forced to split to 2 lists, user facing and not. You may decide to split user facing further. It doesn't make sense to split not user facing further (nor does it make much sense to include non user facing at all - that's what a git log is for).

Reply:
I'm fine omitting internal-only changes entirely from the changelog.

Reply:
I think I'm okay with that, too. Our changelogs are _long_ and I can imagine it being overwhelming upgrading across stable releases.